app/src/org/commcare/models/database/user/models/CaseIndexTable.java

```diff
@@ -112,4 +117,38 @@ public void clearCaseIndices(int recordId) {
         Cursor c = db.query(TABLE_NAME, new String[]{COL_CASE_RECORD_ID}, COL_INDEX_NAME + " = ? AND " + COL_INDEX_TARGET + " =  ?", args, null, null, null);
         return SqlStorage.fillIdWindow(c, COL_CASE_RECORD_ID);
     }
+    /**
+     * Get a list of Case Record id's for cases which index any of a set of provided values
+     *
+     * @param indexName   The name of the index
+     * @param targetValueSet The set of cases targeted by the index
+     * @return An integer array of indexed case record ids
+     */
+    public Vector<Integer> getCasesMatchingValueSet(String indexName, String[] targetValueSet) {
+        String[] args = new String[1 + targetValueSet.length];
+        args[0] = indexName;
+        for(int i =0 ; i < targetValueSet.length ; ++i) {
+            args[i + 1] = targetValueSet[i];
+        }
+        String inSet = getInSet(targetValueSet.length);
+        if (SqlStorage.STORAGE_OUTPUT_DEBUG) {
+            String query = String.format("SELECT %s FROM %s WHERE %s = ? AND %s in " + inSet, COL_CASE_RECORD_ID, TABLE_NAME, COL_INDEX_NAME, COL_INDEX_TARGET);
+            DbUtil.explainSql(db, query, args);
+        }
+        Cursor c = db.query(TABLE_NAME, new String[]{COL_CASE_RECORD_ID}, COL_INDEX_NAME + " = ? AND " + COL_INDEX_TARGET + " in " + inSet, args, null, null, null);
+        return SqlStorage.fillIdWindow(c, COL_CASE_RECORD_ID);
+    }
+
+    public static String getInSet(int number) {
```

Comment:
Nice function, small DRY refactor for future PR:
 https://github.com/dimagi/commcare-core/blob/a2dbf99080e5121e674dc5c809a4f4ea7383954c/src/api/java/org/commcare/api/persistence/SqlHelper.java#L144-L144
https://github.com/dimagi/commcare-core/blob/cd7fc0ef293d1d58c00a8320661142557c6d0258/src/main/java/org/commcare/modern/database/TableBuilder.java#L164-L164